contracts/security/Initializable/InitializableInternal.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.8;
+
+import { AddressUtils } from '../../utils/AddressUtils.sol';
+import { InitializableStorage } from './InitializableStorage.sol';
+
+abstract contract InitializableInternal {
+    using AddressUtils for address;
+    using InitializableStorage for InitializableStorage.Layout;
+
+    event Initialized(uint8 version);
+
+    error Initializable__AlreadyInitialized();
+
+    error Initializable__NotInitializing();
+
+    error Initializable__IsInitializing();
+
+    modifier initializer() {
+        InitializableStorage.Layout storage l = InitializableStorage.layout();
+        bool isTopLevelCall = !l.initializing;
+        bool passed = (isTopLevelCall && l.initialized < 1) ||
+            (!address(this).isContract() && l.initialized == 1);
```

Comment:
These variable declarations use 4 of the 16 available stack positions, which adds limitations to any function that uses the modifier.  `passed` can be removed easily, because it's only used once.  `isTopLevelCall` can be replaced with direct lookups to `!l.initializing` - this will cost more gas, but that's not a significant concern for an initializer function, which is called very rarely.

Reply:
1. I removed `passed`.
2. I think that `isTopLevelCall` is meaningful to know the init function whether is the first call or not.